### DISCUSSION TITLE
Spacy Memory Profiling gives some weird behavior!

### DISCUSSION THREAD

**Discussion by Salekeen** [explosion/spaCy#12206] [Help: Coding & Implementations] [answered]
12    594.9 MiB    516.6 MiB           1       import spacy
    13    594.9 MiB      0.0 MiB           1       import spacy 
    14    836.6 MiB    241.7 MiB           1       nlp_de_sm = spacy.load("de_core_news_sm")
    15   1035.4 MiB    198.8 MiB           1       nlp_de_sm = spacy.load("de_core_news_sm")
    16   1035.4 MiB      0.0 MiB           1       return Response("This is good")

Can someone explain to me why line 15 increases memory?

**Comment by polm** [comment, marked as answer]
`import` is specially implemented in Python to not do anything if it's called a second time. In contrast, `spacy.load` doesn't know if it's been called before, so it loads things from disk and creates new structures in memory. Since you are shadowing the old variable, after garbage collection is run the extra memory from the second call should go away, but there's no guarantee it happens immediately.
